base_view_mode_restricted/models/ir_actions_act_window.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2021 - today Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from itertools import chain
+from odoo import fields, models
+
+
+class IrActionsActWindow(models.Model):
+
+    _inherit = "ir.actions.act_window"
+
+    view_mode_restriction_ids = fields.One2many(
+        "ir.actions.view.mode.restriction",
+        "action_id",
+    )
+
+    def _compute_views(self):

```

Comment:
@lanto-razafindrabe it will be preferable to add original depends to this method

@api.depends('view_ids.view_mode', 'view_mode', 'view_id.type')